sql/moz-fx-data-shared-prod/org_mozilla_ios_derived/legacy_mobile_event_counts/view.sql

```diff
@@ -0,0 +1,65 @@
+WITH extracted AS (
+  SELECT
+    *,
+    date(submission_timestamp) AS submission_date,
+  FROM
+    `mozdata.telemetry.mobile_event`
+  WHERE
+    DATE(submission_timestamp) = "2021-02-01"
+),
+meta AS (
+  SELECT
+    * EXCEPT (events)
+  FROM
+    extracted
+),
+meta_ranked AS (
+  SELECT
+    t AS metadata,
+    row_number() OVER (PARTITION BY client_id ORDER BY submission_timestamp DESC) AS _n
+  FROM
+    meta t
+),
+meta_recent AS (
+  SELECT
+    metadata.*
+  FROM
+    meta_ranked
+  WHERE
+    _n = 1
+),
+unnested AS (
+  SELECT
+    * EXCEPT (events),
+    mozdata.udf.deanonymize_event(event).*
+  FROM
+    extracted,
+    UNNEST(events) AS event
+),
+counts AS (
+  SELECT
+    client_id,
+    submission_date,
+    event_category AS category,
+    event_method AS method,
+    event_object AS object,
+    event_string_value AS string_value,
+    COUNT(DISTINCT document_id) AS value
+  FROM
+    unnested
+  GROUP BY
+    client_id,
+    submission_date,
+    category,
+    method,
+    object,
+    string_value
+)
+SELECT
+  *
+FROM
+  counts
+JOIN
+  meta_recent
```

Comment:
I'm not sure this will work for historical client dimensions. Take, for example, `os_version`. We expect that over the period of 2019-2020 for `os_version` to change significantly as iOS updates and people buy new phones. We won't be able to see this change when using just the latest metadata.

Reply:
This metadata is being done per client per day, so we should expect to see differences in the os_version over time from view. It was my mistake for not pushing the latest version of this view (which is now an incremental query).